src/ArgSortMsg.chpl

```diff
@@ -404,95 +424,47 @@ module ArgSortMsg
-                    st.addEntry(ivname, createSymEntry(iv));
-                }
-                when (DType.Bool) {
-                    var e = toSymEntry(gEnt,bool, nd);
-                    var int_ea = makeDistArray(e.a:int);
-                    var iv = argsortDefault(int_ea, algorithm=algorithm, axis);
-                    st.addEntry(ivname, createSymEntry(iv));
-                }
-                otherwise {
-                    var errorMsg = notImplementedError(pn,gEnt.dtype);
-                    asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-                    return new MsgTuple(errorMsg, MsgType.ERROR);
-                }
-            }
-          }
-          when ObjType.STRINGS {
-            if nd != 1 {
-              const errorMsg = "argsort only supports 1D strings";
-              asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-              return new MsgTuple(errorMsg, MsgType.ERROR);
-            }
-            var strings = getSegString(name, st);
-            // check and throw if over memory limit
-            overMemLimit((8 * strings.size * 8)
-                         + (2 * here.maxTaskPar * numLocales * 2**16 * 8));
-            var iv = strings.argsort();
-            st.addEntry(ivname, createSymEntry(iv));
-          }
-          otherwise {
-              var errorMsg = notImplementedError(pn, objtype: string);
-              asLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
-              return new MsgTuple(errorMsg, MsgType.ERROR);
-          }
-        }
+    @arkouda.instantiateAndRegister
+    proc argsort(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+      where (array_dtype != BigInteger.bigint) && (array_dtype != uint(8))
+    {
+        const name = msgArgs["name"],
+              algoName = msgArgs["algoName"].toScalar(string),
+              algorithm = getSortingAlgoritm(algoName),
+              axis =  msgArgs["axis"].toScalar(int),
+              symEntry = st[msgArgs["name"]]: SymEntry(array_dtype, array_nd),
+              vals = if (array_dtype == bool) then (symEntry.a:int) else (symEntry.a: array_dtype);
+    
+        const iv = argsortDefault(vals, algorithm=algorithm, axis);
+        return st.insert(new shared SymEntry(iv));
+    }
+
+    proc argsort(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+      where array_dtype == BigInteger.bigint
+    {
+        return MsgTuple.error("argsort does not support the bigint dtype");
+    }
+
+    proc argsort(cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int): MsgTuple throws 
+      where (array_dtype == uint(8)) 
+    {
+        return MsgTuple.error("argsort does not support the unit(8) dtype");
```

Comment:
both of these overloads could be combined with:

```chapel
return MsgTuple.error("argsort does not support the %s dtype".format(array_dtype:string));
```

Reply:
Great suggestion!  I made the change.  Thanks.